test/functional/options/defaults_spec.lua

```diff
@@ -328,3 +332,235 @@ describe('XDG-based defaults', function()
     end)
   end)
 end)
+
+
+describe('stdpath()', function()
+  -- Returns an OS friendly version of the unix_path.
+  local function alter_path(unix_path)
+    if os_name() == 'windows' then
+      return string.gsub(unix_path, '/', '\\')
+    else
+      return unix_path
+    end
+  end
+
+  describe('"config"' , function ()
+    it('knows XDG_CONFIG_HOME', function()
+      clear({env={
+        XDG_CONFIG_HOME=alter_path('/home/docwhat/.config')
+      }})
+      eq(alter_path('/home/docwhat/.config/nvim'), funcs.stdpath('config'))
+    end)
+
+    it('handles changes during runtime', function()
+      clear({env={
+        XDG_CONFIG_HOME=alter_path('/home/original')
+      }})
+      eq(alter_path('/home/original/nvim'), funcs.stdpath('config'))
+      nvim('command', 'let $XDG_CONFIG_HOME="/home/new"')
+      eq(alter_path('/home/new/nvim'), funcs.stdpath('config'))
+    end)
+
+    it("doesn't expand $VARIABLES", function()
+      clear({env={
+        XDG_CONFIG_HOME='$VARIABLES',
+        VARIABLES='this-should-not-happen'
+      }})
+      eq(alter_path('$VARIABLES/nvim'), funcs.stdpath('config'))
+    end)
+
+    it("doesn't expand ~/", function()
+      clear({env={
+        XDG_CONFIG_HOME='~/frobnitz'
+      }})
+      eq(alter_path('~/frobnitz/nvim'), funcs.stdpath('config'))
+    end)
+  end)
+
+  describe('"data"' , function ()
+    local appended_dir
+    before_each(function()
+      -- Windows appends 'nvim-data' instead of just 'nvim' to
+      -- prevent collisions due to XDG_CONFIG_HOME and XDG_DATA_HOME
+      -- being the same.
+      if os_name() == 'windows' then
```

Comment:
I don't know what `clear()` actually does.  I'm cargo-culting that bit of code.

I assume it is [this method](https://github.com/neovim/neovim/blob/19d38c4d0f98004e8b2c39821f9bee8ca7e2f5d2/test/functional/helpers.lua#L258) but there are no docs and I'm not skilled enough at Lua to understand what it is doing.

Also, can you confirm that [Busted](http://olivinelabs.com/busted/) is what's being used for the lua tests? It isn't mentioned in `CONTRIBUTING.md`, so I googled for "rspec like lua tests" 😃